### ISSUE TITLE
Custom struct throwing NotSupportedException on getting JsonEncodingId

### ISSUE THREAD

**Issue by OlivierVerhaegen** [OPCFoundation/UA-ModelCompiler#185]
Hi All,

I noticed that the `JsonEncodingId` currently throws an `NotSupportedException` although the class implements `IJsonEncodeable`.
This happens when running `Server.Factory.AddEncodeableTypes(...)`.
This seems weird to me, any explanation?

![image](https://github.com/user-attachments/assets/2000a414-a546-44c2-89de-b93b489cbe0c)

![image](https://github.com/user-attachments/assets/0da0fc5d-1f46-4042-9510-136990d7e395)

Thanks already for the information.

**Comment by opcfoundation-org** [comment]
If the JsonEncodingIds are not defined in the NodeSet then it can't be implemented.
The JsonEncodingIds are optional in 1.05.03 and removed from 1.05.4.
The DataTypeId is used instead.

**Comment by OlivierVerhaegen** [comment]
Hi thank you for the quick response!

I'm currently using Model version 1.4.10 so that would explain why no JsonEncodingIds are available?
How should I implement this, just by updating to 1.05.4?

**Comment by opcfoundation-org** [comment]
You not get an exception in AddEncodeableTypes since there is a try catch block to suppress this error.

Can you explain exactly where the exception is coming from?

**Comment by OlivierVerhaegen** [comment]
I cannot see the insides of AddEncodeableTypes but I only get the exception when running in debug mode.

**Comment by opcfoundation-org** [comment]
Code is here: https://github.com/OPCFoundation/UA-.NETStandard/blob/master/Stack/Opc.Ua.Core/Types/Encoders/EncodeableFactory.cs

Do you have some setting enabled to stop on exception? This can cause the debugger to halt even if an exception is handled.

**Comment by OlivierVerhaegen** [comment]
This seems that the exception is categorized as "User-Unhandled".
When I disable the breakpoint optionf for these kind of exceptions the code runs but weirdly I don't get the error logs defined in the code you mentioned above.
I only get these logs:
![image](https://github.com/user-attachments/assets/eb129e31-6b27-4efc-b475-648d20b894a8)

**Comment by opcfoundation-org** [comment]
opcua.dll means you are using someone else's code base. where did you get the dll from?

**Comment by OlivierVerhaegen** [comment]
I'm just compiling my own project using .NET core on Linux and removed a part of the project name, hence why you only see opcua.dll in the name

**Comment by opcfoundation-org** [comment]
You should not get an error. AddEncodeableTypes suppresses the NotImplementedException and continues.

**Comment by OlivierVerhaegen** [comment]
Yeah I know I've read the code (not only master but also specific to the version I'm using).
I'm using the following nuget package: `<PackageReference Include="OPCFoundation.NetStandard.Opc.Ua.Server" Version="1.5.374.78" />`

.Net Core version 8.0.303 on Ubuntu 20.04.3 LTS
